File: 8_kyu/Fake_Binary.c
#include <stddef.h>

// Transform string "digits"  into string "buffer".
// Replaces all <'5' by '0' all others by '1'.
void fakeBin(const char digits [], char buffer []) {
  size_t index = 0;
  for( ; digits[index]; ++index )
    buffer[index] = digits[index] < '5' ? '0' : '1';
  buffer[index] = '\0';
}

File: 8_kyu/Fake_Binary.rs
// Converts the string to a fake binary string.
// Replaces all chars below '5' to '0' all other chars to '0'.
fn fake_bin(s: &str) -> String {
    s.chars().map(|c| if c < '5' { '0' } else { '1' }).collect()
}
